trpcultivate_phenotypes/tests/src/Kernel/ServiceGenusProjectTest.php

```diff
@@ -128,72 +206,213 @@ public function provideGenusProjectForGenusProjectService() {
    *
    * @param string $scenario
    *   Human-readable text description of the test scenario.
-   * @param string $project_name
-   *   A string, the name or title of a project.
-   * @param array $project_genus
-   *   An array, the genus and species of the organism that will be
-   *   paired with the project. Keyed by 'genus' and 'species'.
+   * @param int $genus_count
+   *   An integer indicating the number of genus to set to the project.
    * @param array $expected
    *   An array of expected values, with the following keys:
    *     - 'project_genus': the expected genus returned by the method
    *     getGenusOfProject().
-   *     - 'alternative_genus': a genus returned after setting a genus.
    *
    * @dataProvider provideGenusProjectForGenusProjectService
    */
-  public function testGenusProjectService($scenario, $project_name, $project_genus, $expected) {
-    // Create the project record.
-    $project_id = $this->chado_connection->insert('1:project')
-      ->fields([
-        'name' => $project_name,
-        'description' => 'A project description',
-      ])
-      ->execute();
+  public function testGenusProjectService($scenario, $genus_count, $expected) {
+
+    for ($i = 0; $i < $genus_count; $i++) {
+      $genus = 'Genus' . ($i + 1);
+
+      $is_set = $this->service_PhenoGenusProject->setGenusToProject($this->project, $genus);
+      $this->assertTrue($is_set, 'Project Genus Service failed to set a genus to project in scenario: ' . $scenario);
+    }
+
+    $set_genus = $this->service_PhenoGenusProject->getGenusOfProject($this->project);
+
+    $this->assertEquals(
+      count($set_genus),
+      count($expected['project_genus']),
+      'The number of genus set does no match expected genus count in scenario: ' . $scenario
+    );
```

Comment:
I would use assertCount here with the first param being $genus_count and the second being $set_genus. This will give more feedback on failure.